### ISSUE TITLE
UI refactoring

### ISSUE THREAD

**Issue by turtlegarden** [aleiepure/devtoolbox#39]
**Is your feature request related to a problem? Please describe.**
Currently, the UI is very packed and does not work well with adaptivity. The header bar also contains very many items, which can be confusing.

**Describe the solution you'd like**
Using `AdwNavigationSplitView` should allow nested sidebars, or a three pane view to be added. I think a three pane view would be best for this application instead of the header bar.

**Screenshots**
![Skärmbild från 2023-11-19 18-49-13](https://github.com/aleiepure/devtoolbox/assets/52264162/e051c842-a778-440b-9ec8-f6749f73adfb)

Mockup:
![Skärmbild från 2023-11-20 10-11-06](https://github.com/aleiepure/devtoolbox/assets/52264162/7a05eef7-4c6e-499e-9577-fa7a99fc955f)

**Comment by aleiepure** [comment]
Thank you for your feedback. 
I am working on porting the app to the new libadwaita widgets and adding some new features in a separate branch. 
As part of this effort I have already replaced the tabs with a navigation Split View and moved all tools to a categorized sidebar list. I have also implemented a search function to easily find the tools in a long list and the ability to mark tools as favorites. 
Before merging it I need to go over the individual utilities and update my custom widgets to the new ones provided by the new adwaita version. 
Unfortunately the progress is very slow due to university and other IRL stuff.
